File: backend/VirtualLibrary.Domain/Entities/Book.cs
namespace VirtualLibrary.Domain.Entities;

public class Book
{
    public string ISBN { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Author { get; set; } = string.Empty;
    public string Publisher { get; set; } = string.Empty;
    public int? PublicationYear { get; set; }
    public string Description { get; set; } = string.Empty;
    public string CoverImageUrl { get; set; } = string.Empty;
    public List<string> Categories { get; set; } = new();
}

File: ios/VirtualLibrary/Models/Book.swift
import Foundation

struct Book: Identifiable, Codable {
    var id: String { isbn }
    let isbn: String
    let title: String
    let author: String
    let publisher: String
    let publicationYear: Int?
    let description: String
    let coverImageUrl: String
    let categories: [String]
    
    enum CodingKeys: String, CodingKey {
        case isbn
        case title
        case author
        case publisher
        case publicationYear
        case description
        case coverImageUrl
        case categories
    }
}
